src/util.py

```diff
@@ -340,7 +340,23 @@ def process_config(config):
                                      f'(current length {hash_length}).')
                 algorithm_params[alg["name"]][params_hash] = run_dict
 
-    return config, datasets, out_dir, algorithm_params, algorithm_directed
+
+    analysis_params = config["analysis"] if "analysis" in config else {}
+    ml_params = analysis_params["ml"] if "ml" in analysis_params else {}
+    
+    pca_params = {
+        "components": ml_params["components"] if "components" in ml_params else 2,
+        "labels": ml_params["labels"] if "labels" in ml_params else True,
+    }
+
+    hac_params = {
+        "linkage": ml_params["linkage"] if "linkage" in ml_params else 'ward',
+        "metric": ml_params["metric"] if "metric" in ml_params else 'euclidean',
+    }
```

Comment:
Do we need to set default values here or do we let the default values in the functions specify the behavior? I'm unsure. We wouldn't want the hard-coded defaults to diverge though.